tools/dynamicdatamonitor/DynamicDataMonitor.cpp

```diff
@@ -0,0 +1,181 @@
+
+    OpenDDS::DCPS::Service_Participant* service = TheServiceParticipant;
+
+    // Create DomainParticipant
+    DDS::DomainParticipant_var participant =
+      dpf->create_participant(domainid,
+                              PARTICIPANT_QOS_DEFAULT,
+                              0,
+                              OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+    if (!participant) {
+      ACE_ERROR_RETURN((LM_ERROR,
+                        ACE_TEXT("ERROR: %N:%l: main() -")
+                        ACE_TEXT(" create_participant failed!\n")),
+                       -1);
+    }
+    using namespace OpenDDS::DCPS;
+
+    {
+      DDS::Topic_var topic =
+        service->create_typeless_topic(participant,
+                                       topic_name.c_str(),
+                                       type_name.c_str(),
+                                       true,
+                                       TOPIC_QOS_DEFAULT,
+                                       0,
+                                       OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+      if (!topic) {
+        ACE_ERROR_RETURN((LM_ERROR,
+                          ACE_TEXT("ERROR: %N:%l: main() -")
+                          ACE_TEXT(" create_topic failed!\n")),
+                         -1);
+      }
+
+      // ACE_Time_Value wait_time(60, 0);
+
+      RcHandle<TestRecorderListener> recorder_listener = make_rch<TestRecorderListener> ();
+
+      DDS::SubscriberQos sub_qos;
+      participant->get_default_subscriber_qos(sub_qos);
+
+      DDS::DataReaderQos dr_qos = service->initial_DataReaderQos();
+      dr_qos.representation.value.length(1);
+      dr_qos.representation.value[0] = DDS::XCDR2_DATA_REPRESENTATION;
+      // Create Recorder
+      OpenDDS::DCPS::Recorder_var recorder =
+        service->create_recorder(participant,
+                                 topic.in(),
+                                 sub_qos,
+                                 dr_qos,
+                                 recorder_listener);
+
+      if (!recorder.in()) {
+        ACE_ERROR_RETURN((LM_ERROR,
+                          ACE_TEXT("ERROR: %N:%l: main() -")
+                          ACE_TEXT(" create_recorder failed!\n")),
+                         -1);
+      }
+
+      int i;
+      while (1) {
+        ++i;
+      }
+      // wait until the writer disconnnects
+      // if (recorder_listener->wait(wait_time) == -1) {
+      //   ACE_ERROR_RETURN((LM_ERROR,
+      //                     ACE_TEXT("ERROR: %N:%l: main() -")
+      //                     ACE_TEXT(" recorder timeout!\n")),
+      //                    -1);
+      // }
+      // ret_val = recorder_listener->ret_val_;
+      // service->delete_recorder(recorder);
+    }
+    // participant->delete_contained_entities();
+    // dpf->delete_participant(participant);
+    // TheServiceParticipant->shutdown();
+  } catch (const CORBA::Exception& e) {
+    e._tao_print_exception("Exception caught in main():");
+    return -1;
```

Comment:
Please don't return negative numbers from `main`. They can't be properly checked from a shell.